Three added settings, plus some smaller fixes.

zipponator opened the pull request.

zipponator commented:
Hm ... I'm leaving for vacation now (obviously! :D) so won't have time to investigate the failed build thing for several weeks :/
I should add that the tests passed on my machine ... :)

shaunburdick commented:
Coveralls is complaining that the test coverage lowered (ie you added code but didn't add tests to cover it)

zipponator commented:
@shaunburdick: hi! 
Rebased this on master.
Is there anything you want me to add/change here?
I'd be happy to add some tests for the stuff I added, but as far as I can tell, that's not very easy to do ... or is there a way to test things in the actual bot.js behaviour?
